language/move-core/types/src/language_storage.rs

```diff
@@ -157,3 +157,143 @@ impl Display for ResourceKey {
         write!(f, "0x{}/{}", self.address.short_str_lossless(), self.type_)
     }
 }
+
+// =================================================================================================
+// Type definition with free type variables
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq, PartialOrd, Ord)]
+pub enum Type {
+    Bool,
+    U8,
+    U64,
+    U128,
+    Address,
+    Signer,
+    Vector(Box<Type>),
+    Struct(StructType),
+    TypeArg(usize),
```

Comment:
I'm confused why there are type variables here at all? When does this ever happen for storage?

Reply:
Yea this is really a bad location... Removed this confusing piece entirely.